eth/downloader/queue.go

```diff
@@ -49,35 +56,73 @@ type fetchRequest struct {
 	Time    time.Time           // Time when the request was made
 }
 
+// fetchResult is the assembly collecting partial results from potentially more
+// than one fetcher routines, until all outstanding retrievals complete and the
+// result as a whole can be processed.
+type fetchResult struct {
+	Pending int // Number of data fetches still pending
+
+	Header       *types.Header
+	Uncles       []*types.Header
+	Transactions types.Transactions
+	Receipts     types.Receipts
+}
+
 // queue represents hashes that are either need fetching or are being fetched
 type queue struct {
+	mode          SyncMode // Synchronisation mode to decide on the block parts to schedule for fetching
+	fastSyncPivot uint64   // Block number where the fast sync pivots into archive synchronisation mode
```

Comment:
Yes, originally I called it threshold, but wanted something shorter. Calling it `fastSyncThreshold` is a way too long name in Go naming conventions and simply `threshold` is not descriptive enough to figure what it's used for. If you really dislike it I'll change it, but I'd rather use something short and sweet.